tests/output.rs

```diff
@@ -44,62 +44,106 @@ struct ExpectedMatch {
     pub range: Range,
 }
 
-impl ExpectedMatch {
-    pub fn expected_output_text(&self) -> String {
-        self.lines
-            .iter()
-            .enumerate()
-            .map(|(line_index, line)| {
-                format!(
-                    "{}:{}:{}",
-                    self.relative_file_path,
-                    self.range.start.line + line_index + 1,
-                    line
-                )
-            })
-            .collect::<Vec<_>>()
-            .join("\n")
-    }
-}
-
 const MATCH_OUTPUT_LINE_REGEX_STR: &'static str = r#"(^|\n).+:\d+:"#;
 
-fn predicate_from_expected_matches(expected_matches: &[ExpectedMatch]) -> BoxPredicate<str> {
+fn predicate_from_expected_matches(
+    expected_matches: &[ExpectedMatch],
+    output_mode: OutputMode,
+) -> BoxPredicate<str> {
     expected_matches.into_iter().fold(
         BoxPredicate::new(
             predicate::str::is_match(MATCH_OUTPUT_LINE_REGEX_STR)
                 .unwrap()
-                .count(
-                    expected_matches
-                        .into_iter()
-                        .map(|expected_match| expected_match.lines.len())
-                        .sum(),
-                ),
+                .count(output_mode.get_expected_total_number_of_match_lines(expected_matches)),
```

Comment:
...ya something felt right when writing it to put this and the other method on `OutputMode` but when reading it I'm not so sure